domain-models-runtime/src/main/java/org/folio/rest/persist/PostgresClient.java

```diff
@@ -2801,31 +2802,92 @@ public void selectSingle(AsyncResult<SQLConnection> conn, String sql, Handler<As
    * @param replyHandler  The query result or the failure.
    */
   public void selectSingle(String sql, JsonArray params, Handler<AsyncResult<JsonArray>> replyHandler) {
```

Comment:
This is inconsistent.
`select(String sql, JsonArray params, Handler<AsyncResult<RowSet<Row>>> replyHandler)`
returns RowSet<Row> but selectSingle returns JsonArray, I expect Row.

Reply:
ok. Makes sense. Will change it to return Row

Reply:
If the API returns not even a single row.. What should we do?
1. throw exception
2. success with fake a Row with no columns (that's what it does at the moment).
3. success with null Row

Reply:
It could return an empty Optional and have the consumer handle the not found exception.

Reply:
The javadoc of the old selectSingle says:

    Run a select query and return the first record, or null if there is no result.

I think we can stay with the old solution. A null check is more natural than a column count check.

Reply:
Yes. Thx